backend/tests/e2e/e2e_utils.py

```
@@ -0,0 +1,311 @@
+
+        async def mock_call_tool(self, server_name: str, tool_name: str, parameters, session_id: str, stage_execution_id=None):
+            """Override call_tool to use our mock sessions."""
+            # Ensure our mock sessions are available
+            self.sessions = mock_sessions.copy()
+            # Call the original method which will now use our mock sessions
+            return await original_call_tool(self, server_name, tool_name, parameters, session_id, stage_execution_id)
+
+        return mock_list_tools, mock_call_tool
+
+    @staticmethod
+    def create_simple_kubernetes_mcp_config(command_args: List[str] = ["test"], instructions: str = "Test server") -> Dict[str, Any]:
```

Ruff (0.12.2)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)